### ISSUE TITLE
using cvxopt to optimize with thousands of variables, constraints and bounds 

### ISSUE THREAD

**Issue by Lizitheking** [cvxopt/cvxopt#155]
I'm trying to construct a portfolio using CVXOPT. The objective function is linear but there are thousands of variables, constraints, a lot of which are nonlinear. I wonder how to use CVXOPT to solve this particular problem. The difficulty I'm having with is twofold. It's not a linear programming and it's not a quadratic either--it's a non-linear programming. Secondly, some of the the large number of constraints are non-linear. In fact, they are cross terms like x1*x2>=0, x3*x7>=0 and so forth. I was kindly suggested to use GEKKO, but I still haven't been able to figure out how to write thousands of linear constraints in a matrix form and other thousands of non-linear constraints in the form of m.Equation at the same time. Any pointer would be very much appricated! Thank you!

**Comment by martinandersen** [comment]
A constraint of the form x1x2>=0 is nonconvex — you cannot use CVXOPT for this problem. However, if x1 and x2 are also required to be nonnegative, then x1x2>=0 is equivalent to the nonnegativity constraints x1>=0, x2>=0, which can be handled by CVXOPT.

**Comment by Lizitheking** [comment]
Yes, I noticed that CVXOPT is only meant to solve convex optimization. Actually, this was a non-linear optimization because some of the constraints can not reduced to linear equations. When I say I want some of the constraints to be looking like x1*x2>=0, I mean they are either both non-postive or both non-negative. This is in fact a very hard numberical opitimization problem that no exisiting packckage in python is consistently capable of solving this optimization on a large scale, menaning the number of variables are in the thousands. If the nunber of variables is just a few dozens, then it couldn't be easily solved in Excel. I've contaced a dozens of mathematicians or practionners who specialize in numberical optimization and nobody could give it a straight answer and all suugested that I resort to use C++ or Matlab, the more traditional programming language to solve it. None of  the tough numerical real world optimization is done in Python but they are all in C++ and Matlab.
